extras/travis_build.sh

```diff
@@ -205,26 +212,55 @@ fi  # End of Static code analysis.
 
 cd "$NEST_VPATH"
 cp ../examples/sli/nestrc.sli ~/.nestrc
+# Fix for the Open-mpi slot error when using Open-mpi>3.0
+if [[ "$OSTYPE" == "darwin"* ]] ; then
+    sed -i -e 's/mpirun -np/mpirun --oversubscribe -np/g' ~/.nestrc
+fi
 
 echo
 echo "+ + + + + + + + + + + + + + + + + + + + + + + + + + + + + + + + + + + + + + + +"
 echo "+               C O N F I G U R E   N E S T   B U I L D                       +"
 echo "+ + + + + + + + + + + + + + + + + + + + + + + + + + + + + + + + + + + + + + + +"
 echo "MSGBLD0230: Configuring CMake."
-cmake \
-  -DCMAKE_INSTALL_PREFIX="$NEST_RESULT" \
-  -Dwith-optimize=ON \
-  -Dwith-warning=ON \
-  -Dwith-boost=ON \
-  $CONFIGURE_THREADING \
-  $CONFIGURE_MPI \
-  $CONFIGURE_PYTHON \
-  $CONFIGURE_MUSIC \
-  $CONFIGURE_GSL \
-  $CONFIGURE_LTDL \
-  $CONFIGURE_READLINE \
-  $CONFIGURE_LIBNEUROSIM \
-  ..
+if [[ "$OSTYPE" == "darwin"* ]] ; then
+    cmake \
+      -DCMAKE_INSTALL_PREFIX="$NEST_RESULT" \
+      -Dwith-optimize=ON \
+      -Dwith-warning=ON \
+      -Dwith-boost=OFF \
+      -DCMAKE_C_COMPILER=/usr/local/opt/gcc/bin/gcc-8 \
+      -DOpenMP_C_FLAGS="-fopenmp -I/usr/local/opt/libomp/include" \
+      -DOpenMP_C_LIB_NAMES="omp" \
+      -DOpenMP_omp_LIBRARY=/usr/local/opt/libomp \
+      -DCMAKE_CXX_COMPILER=/usr/local/opt/gcc/bin/g++-8 \
+      -DOpenMP_CXX_FLAGS="-fopenmp -I/usr/local/opt/libomp/include" \
+      -DOpenMP_CXX_LIB_NAMES="omp" \
+      -D_GLIBCXX_USE_CXX11_ABI=0 \
+      $CONFIGURE_THREADING \
```

Comment:
I think it would be cleaner if (instead of having the `if` on the `OSTYPE`) we introduced a `CONFIGURE_BOOST` variable, which we could set depending on the OS we're running on, and a variable `CONFIGURE_MISC`, which could contain all the OsX specific compiler and library settings and be empty otherwise. With these in place, we could go back to the single configure line above and have less code duplication.

Reply:
@jougs : Thanks for the review and valuable suggestions. The idea of having extra `CONFIGURE ` variables for BOOST and OSX related compiler and library settings very much makes sense.
I just made the necessary changes.